sys/Makefile.dep

```diff
@@ -602,6 +602,10 @@ ifneq (,$(filter sock_dns,$(USEMODULE)))
   USEMODULE += posix_headers
 endif
 
+ifneq (,$(filter sock_dns_mock,$(USEMODULE)))
+  USEMODULE += dns
+endif
```

Comment:
If I am not mistaken, this dependency might become unnecessary with the `#if defined(MODULE_DNS)`s removed

Reply:
The problem is that compilation then fails if the `sock` module is not used

```
In file included from /home/benpicco/dev/RIOT/sys/include/net/sock/dns.h:33,
                 from /home/benpicco/dev/RIOT/sys/include/net/dns.h:24,
                 from /home/benpicco/dev/RIOT/sys/net/netutils/util.c:24:
/home/benpicco/dev/RIOT/sys/include/net/sock/udp.h:788:10: fatal error: sock_types.h: No such file or directory
  788 | #include "sock_types.h"
```

Reply:
Mh, so similar problem to `sock_dtls`... but let's not stall this any further and merge as is.